src/openlifu/util/units.py

```diff
@@ -241,3 +241,40 @@ def get_ndgrid_from_arr(data_arr: Dataset) -> np.ndarray:
     ndgrid = np.stack(np.meshgrid(*all_coord, indexing="ij"), axis=-1)
 
     return ndgrid
+
+
+def get_ndgrid_from_coords(coords_arr: Coordinates) -> np.ndarray:
+    """
+    Creates a ndgrid from xarray.coordinates.
+
+    Args:
+        coords : xarray.Coordinates
+
+    Returns:
+        ndgrid: The ndgrid from the Coordinates.
+    """
+    all_coord = []
+    for coord_key in coords_arr.dims:
+        if 'units' in coords_arr[coord_key].attrs:
+            all_coord += [coords_arr[coord_key].data]
+    ndgrid = np.stack(np.meshgrid(*all_coord, indexing="ij"), axis=-1)
+
+    return ndgrid
+
+
+def convert_transform(matrix: np.ndarray, units: str, tgt_units: str) -> np.ndarray:
+    """Given a transform matrix in some units, convert it to another units.
+
+    Args:
+        matrix: 4x4 np.ndarray
+            affine transform matrix
+        units: str
+            units of the coordinate space on which the provided transform matrix operates
+
+    Returns: 4x4 affine transform matrix
+    """
```

Comment:
- incomplete docstring: `tgt_units` is missing
- duplicates code from `Trasnducer.convert_transform`. The latter should be made to use this, if we really need this function here in utils.
- This function should be unit tested. Can pretty much copy the existing unit test for `Trasnducer.convert_transform`

Reply:
Yes I saw it, but because the transform is now independent from a transducer I found it weird that this method was tied to a transducer.  I removed the one I created,